### PR TITLE
feat(amf): Logging improvements at function level

### LINKED ISSUE #12277: [NGAP][AMF]: To support profiling the major functions
Hi Priya,

Please uniform all the relevant functions with the below OAILOG macros to support for profiling and aid in debugging the issues.

OAILOG_FUNC_IN
OAILOG_FUNC_RETURN

Basic idea is to enable basic profiling with these macros to identify and optimize the heavily functions.

Regards,
Kaleem

### PR DESCRIPTION
**Pull request by priya-wavelabs**
## Summary

Modified the functions entry and exit points under OAI_LOG for tracing.

## Test Plan
Verified with unit test.
Screenshot 2022-05-09 143338

Pcap: 
image

Logs: 
[mme02.log](https://github.com/magma/magma/files/8650135/mme02.log)

## Additional Information

- [ ] This change is backwards-breaking

### CODE REVIEW

File: lte/gateway/c/core/oai/tasks/amf/Registration.cpp [hunk 1]
```diff
@@ -691,6 +695,7 @@ int amf_send_registration_accept(amf_context_t* amf_context) {
 
 static int registration_accept_t3550_handler(zloop_t* loop, int timer_id,
                                              void* arg) {
+  OAILOG_FUNC_IN(LOG_NAS_AMF);
```
Comment (rsarwad) [inline lte/gateway/c/core/oai/tasks/amf/Registration.cpp:691]: In this function, at some places. LOG_AMF_APP is used. For same function, module name should be same

File: lte/gateway/c/core/oai/tasks/amf/amf_app_ue_context.cpp [hunk 2]
```diff
@@ -261,7 +267,7 @@ amf_context_t* amf_context_get(const amf_ue_ngap_id_t ue_id) {
     }
     OAILOG_DEBUG(LOG_NAS_AMF, "Stored UE id " AMF_UE_NGAP_ID_FMT " \n", ue_id);
   }
-  return amf_context_p;
+  OAILOG_FUNC_RETURN(LOG_AMF_APP, amf_context_p);
```
Comment (rsarwad) [inline lte/gateway/c/core/oai/tasks/amf/amf_app_ue_context.cpp:270]: Module name is not same in this fun(0. 
General comment to have same module name in one function

File: lte/gateway/c/core/oai/tasks/amf/Registration.cpp (hunk 1 above)
Comment (priya-wavelabs) [reply lte/gateway/c/core/oai/tasks/amf/Registration.cpp:691]: Comment addressed.

File: lte/gateway/c/core/oai/tasks/amf/amf_app_ue_context.cpp (hunk 2 above)
Comment (priya-wavelabs) [reply lte/gateway/c/core/oai/tasks/amf/amf_app_ue_context.cpp:270]: Comment addressed.

**Review by ganeshg87 — APPROVED** [review]
lgtm

File: lte/gateway/c/core/oai/tasks/amf/amf_data.cpp [hunk 3]
```diff
@@ -110,6 +117,7 @@ void nas_amf_smc_proc_t::amf_ctx_clear_security(amf_context_t* ctxt) {
   OAILOG_DEBUG(
       LOG_NAS_AMF, "ue_id= " AMF_UE_NGAP_ID_FMT " cleared security context \n",
```
Comment (rsarwad) [inline lte/gateway/c/core/oai/tasks/amf/amf_data.cpp]: Change the module name

**Review by rsarwad — APPROVED** [review]
LGTM

File: lte/gateway/c/core/oai/tasks/amf/amf_data.cpp (hunk 3 above)
Comment (priya-wavelabs) [reply lte/gateway/c/core/oai/tasks/amf/amf_data.cpp]: Comment addressed.

**Comment by panyogesh** [comment]
LGTM

**Pull request merged**